src/coreclr/jit/gentree.cpp

```diff
@@ -16180,7 +16180,7 @@ bool GenTree::DefinesLocal(Compiler* comp, GenTreeLclVarCommon** pLclVarTree, bo
 // Returns true if this GenTree defines a result which is based on the address of a local.
 bool GenTree::DefinesLocalAddr(Compiler* comp, unsigned width, GenTreeLclVarCommon** pLclVarTree, bool* pIsEntire)
 {
-    if (OperGet() == GT_ADDR || OperGet() == GT_LCL_VAR_ADDR)
+    if (OperIs(GT_ADDR, GT_LCL_VAR_ADDR, GT_LCL_FLD_ADDR))
```

Comment:
Nit: [this](https://github.com/dotnet/runtime/blob/c82e8b2742c4e239496b45089d0575bb8f447d64/src/coreclr/jit/lclmorph.cpp#L724-L728) local morph comment is now outdated.

Reply:
Do you know why `DefinesLocalAddr` was not recognizing the pattern in the first place? Since it supports communicating "entirety" it seems pretty uncontroversial.

Reply:
> Do you know why DefinesLocalAddr was not recognizing the pattern in the first place?

Not really I suppose. If I were to guess, it is the result of `LCL_FLD` being a somewhat late addition, and the original version of the function only recognizing `ADDR(LCL_VAR)`.

I do believe are no correctness dependencies on it not recognizing `LCL_FLD_ADDR`.

(Side note: the contract of `DefinesLocalAddr` is that it must recognize a superset of addresses `LocalAddressVisitor` recognizes, absence of `FLD_ADDR` seemingly violates this, but not actually, because it does recognize `ADDR(LCL_FLD)`, and in LIR we rely on use-def flags set by the front-end)